src/prefect/cli/deploy.py

```diff
@@ -591,51 +592,56 @@ async def _run_single_deploy(
+            console=app.console,
         ):
-            app.console.print(
-                "[red]Cancelled saving deployment configuration"
-                f" '{deploy_config_before_templating.get('name')}' to the prefect.yaml"
-                " file.[/red]"
-            )
-        else:
-            _save_deployment_to_prefect_file(
-                deploy_config_before_templating,
-                build_steps=build_steps or None,
-                push_steps=push_steps or None,
-                pull_steps=pull_steps or None,
-                triggers=trigger_specs or None,
+            matching_deployment_exists = (
+                _check_for_matching_deployment_name_and_entrypoint_in_prefect_file(
+                    deploy_config=deploy_config_before_templating
+                )
             )
-            app.console.print(
+            if matching_deployment_exists and not confirm(
                 (
-                    "\n[green]Deployment configuration saved to prefect.yaml![/] You"
-                    " can now deploy using this deployment configuration"
-                    " with:\n\n\t[blue]$ prefect deploy -n"
-                    f" {deploy_config['name']}[/]\n\nYou can also make changes to this"
-                    " deployment configuration by making changes to the prefect.yaml"
-                    " file."
+                    "Found existing deployment configuration with name:"
+                    f" [yellow]{deploy_config_before_templating.get('name')}[/yellow]"
+                    " and entrypoint:"
+                    f" [yellow]{deploy_config_before_templating.get('entrypoint')}[/yellow]"
+                    " in the [yellow]prefect.yaml[/yellow] file. Would you like to"
+                    " overwrite that entry?"
                 ),
-            )
+            ):
+                app.console.print(
+                    "[red]Cancelled saving deployment configuration"
+                    f" '{deploy_config_before_templating.get('name')}' to the"
+                    " prefect.yaml file.[/red]"
+                )
+            else:
+                _save_deployment_to_prefect_file(
+                    deploy_config_before_templating,
+                    build_steps=build_steps or None,
+                    push_steps=push_steps or None,
+                    pull_steps=pull_steps or None,
+                    triggers=trigger_specs or None,
+                )
+                app.console.print(
+                    (
+                        "\n[green]Deployment configuration saved to prefect.yaml![/]"
+                        " You can now deploy using this deployment configuration"
+                        " with:\n\n\t[blue]$ prefect deploy -n"
+                        f" {deploy_config['name']}[/]\n\nYou can also make changes to"
+                        " this deployment configuration by making changes to the"
+                        " prefect.yaml file."
+                    ),
+                )
```

Comment:
This was just moved down, but hasn't changed other than being one indentation over